src/Compilers/CSharp/Test/Syntax/Parsing/CollectionLiteralParsingTests.cs

```diff
@@ -12821,4 +12777,3988 @@ public void AttributeOnTopLevelFunction5()
+                {
+                    N(SyntaxKind.IdentifierToken, "f");
+                }
+            }
+            M(SyntaxKind.ColonToken);
+            M(SyntaxKind.IdentifierName);
+            {
+                M(SyntaxKind.IdentifierToken);
+            }
+        }
+        EOF();
+    }
+
+    [Fact]
+    public void LambdaAttributeVersusCollectionLookahead2A()
+    {
+        UsingExpression("[A][B](C, D) ? e : f");
+
+        N(SyntaxKind.ConditionalExpression);
+        {
+            N(SyntaxKind.InvocationExpression);
+            {
+                N(SyntaxKind.ElementAccessExpression);
+                {
+                    N(SyntaxKind.CollectionExpression);
+                    {
+                        N(SyntaxKind.OpenBracketToken);
+                        N(SyntaxKind.ExpressionElement);
+                        {
+                            N(SyntaxKind.IdentifierName);
+                            {
+                                N(SyntaxKind.IdentifierToken, "A");
+                            }
+                        }
+                        N(SyntaxKind.CloseBracketToken);
+                    }
+                    N(SyntaxKind.BracketedArgumentList);
+                    {
+                        N(SyntaxKind.OpenBracketToken);
+                        N(SyntaxKind.Argument);
+                        {
+                            N(SyntaxKind.IdentifierName);
+                            {
+                                N(SyntaxKind.IdentifierToken, "B");
+                            }
+                        }
+                        N(SyntaxKind.CloseBracketToken);
+                    }
+                }
+                N(SyntaxKind.ArgumentList);
+                {
+                    N(SyntaxKind.OpenParenToken);
+                    N(SyntaxKind.Argument);
+                    {
+                        N(SyntaxKind.IdentifierName);
+                        {
+                            N(SyntaxKind.IdentifierToken, "C");
+                        }
+                    }
+                    N(SyntaxKind.CommaToken);
+                    N(SyntaxKind.Argument);
+                    {
+                        N(SyntaxKind.IdentifierName);
+                        {
+                            N(SyntaxKind.IdentifierToken, "D");
+                        }
+                    }
+                    N(SyntaxKind.CloseParenToken);
+                }
+            }
+            N(SyntaxKind.QuestionToken);
+            N(SyntaxKind.IdentifierName);
+            {
+                N(SyntaxKind.IdentifierToken, "e");
+            }
+            N(SyntaxKind.ColonToken);
+            N(SyntaxKind.IdentifierName);
+            {
+                N(SyntaxKind.IdentifierToken, "f");
+            }
+        }
+        EOF();
+    }
+
+    [Fact]
+    public void LambdaAttributeVersusCollectionLookahead3()
+    {
+        UsingExpression("[A][B] (C, D)? (e) => f");
```

Comment:
It doesn't feel right to me that this case is considered a lambda expression but if we took the parens off of `e` (as in `LambdaAttributeVersusCollectionLookahead2`) it would be considered an incomplete conditional expression. It seems like they should both be treated as lambdas.

Reply:
Lambdas can only have attributes iff they are parenthesized :-)

Reply:
Just to make sure, are we expecting meaningfully different parses for the following?

`[A][B] (C, D)? (e) => f : g`
`[A][B] (C, D)? e => f : g`

Reply:
I see what you're saying. 

It looks like, technically, we get the first wrong (since we view the ? Not as a ternary, but as part of the return type). My general view though is that we are so into the extreme of ambiguity that I care little :-D

It was also clear in this work that we were already getting things wrong with lambdas and ternaries.  We didn't even notice till now, so my career about these crazier and crazier cases is low :-)

Reply:
Works for me.